upgrade.go

```diff
@@ -396,6 +399,12 @@ func (b *providerUpgradeBuilder) checkProviderUpgradePreviewOnly(t *testing.T) {
 	info := b.newProviderUpgradeInfo(t)
 	t.Logf("Baseline provider version: %s", b.baselineVersion)
 
+	// Skip if state not yet created
+	if _, err := os.Stat(info.stateFile); os.IsNotExist(err) {
+		t.Skipf("No pre-recorded state found for %s. Run in 'snapshot' mode to capture the current state produced by the current provider version.", b.baselineVersion)
```

Comment:
@t0yv0 I'm not 100% sure what the procedure is for actually creating a snapshot. If you can point me in the right direction for a playbook or something similar, we could link to that.

Reply:
It's in the README. Run the test with `--provider-snapshot` flag.